### PR TITLE
[fuzz] Fix bugs in HPACK fuzz test

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali asraa@google.com

Commit Message: Fixes test issues in HPACK fuzz test
Additional Message:
* Use after free because nghttp2_nv object has pointers to the underlying strings and copying them resulted in a use after free when the copy was used after the original was destroyed
* Fixed sorting issues and tested leading/trailing whitespace headers (I can no longer reproduce an issue I saw where a null byte appeared after decoding whitespace, maybe the former fix fixed this)

Risk Level: Low
Testing: Added regression tests and cases for whitespace headers
Fixes
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=28880
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=28869

### CODE REVIEW

**Review by htuch — APPROVED** [review]
LGTM, thanks!

File: test/common/http/http2/hpack_fuzz_test.cc [hunk 1]
```diff
@@ -90,10 +96,15 @@ std::vector<nghttp2_nv> decodeHeaders(nghttp2_hd_inflater* inflater,
...
-    return a_str.compare(b_str);
+  inline bool operator()(const nghttp2_nv& a, const nghttp2_nv& b) const {
+    absl::string_view a_name(reinterpret_cast<char*>(a.name), a.namelen);
+    absl::string_view b_name(reinterpret_cast<char*>(b.name), b.namelen);
+    if (a_name != b_name) {
+      return a_name < b_name;
+    }
+    absl::string_view a_val(reinterpret_cast<char*>(a.value), a.valuelen);
+    absl::string_view b_val(reinterpret_cast<char*>(b.value), b.valuelen);
+    return b_val > a_val;
```
Comment (htuch) [inline test/common/http/http2/hpack_fuzz_test.cc:107]: How come this isn't `return a_val < b_val`? Isn't the idea to get lexicographical ordering?

Comment (asraa) [reply test/common/http/http2/hpack_fuzz_test.cc:107]: Hmmmm. I'm not sure if I'm reading the comment correctly, but that is what I'm returning (string_view's `b_val > a_val` returns `a_val < b_val`.) Switched ordering though

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**